src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,620 +53,652 @@ const defaultProps = {
-        this.scrollToIndex = this.scrollToIndex.bind(this);
-        this.selectRow = this.selectRow.bind(this);
-        this.selectFocusedOption = this.selectFocusedOption.bind(this);
-        this.addToSelection = this.addToSelection.bind(this);
-        this.updateSearchValue = this.updateSearchValue.bind(this);
-        this.incrementPage = this.incrementPage.bind(this);
-        this.sliceSections = this.sliceSections.bind(this);
-        this.calculateAllVisibleOptionsCount = this.calculateAllVisibleOptionsCount.bind(this);
-        this.handleFocusIn = this.handleFocusIn.bind(this);
-        this.handleFocusOut = this.handleFocusOut.bind(this);
-        this.debouncedUpdateSearchValue = _.debounce(this.updateSearchValue, CONST.TIMING.SEARCH_OPTION_LIST_DEBOUNCE_TIME);
-        this.relatedTarget = null;
-        this.accessibilityRoles = _.values(CONST.ROLE);
-        this.isWebOrDesktop = [CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform());
-
-        const allOptions = this.flattenSections();
-        const sections = this.sliceSections();
-        const focusedIndex = this.getInitiallyFocusedIndex(allOptions);
-        this.focusedOption = allOptions[focusedIndex];
-
-        this.state = {
-            sections,
-            allOptions,
-            focusedIndex,
-            shouldDisableRowSelection: false,
-            errorMessage: '',
-            paginationPage: 1,
-            disableEnterShortCut: false,
-            value: '',
-        };
-    }
-
-    componentDidMount() {
-        this.subscribeToEnterShortcut();
-        this.subscribeToCtrlEnterShortcut();
-        this.subscribeActiveElement();
-
-        if (this.props.isFocused && this.props.autoFocus && this.textInput) {
-            this.focusTimeout = setTimeout(() => {
-                this.textInput.focus();
-            }, CONST.ANIMATED_TRANSITION);
-        }
-
-        this.scrollToIndex(this.props.selectedOptions.length ? 0 : this.state.focusedIndex, false);
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.disableEnterShortCut !== this.state.disableEnterShortCut) {
-            // Unregister the shortcut before registering a new one to avoid lingering shortcut listener
-            this.unsubscribeEnter();
-            if (!this.state.disableEnterShortCut) {
-                this.subscribeToEnterShortcut();
+function BaseOptionsSelector(props) {
+    const isFocused = useIsFocused();
+    const {translate} = useLocalize();
+    const themeStyles = useThemeStyles();
+
+    const getInitiallyFocusedIndex = useCallback(
+        (allOptions) => {
+            let defaultIndex;
+            if (props.shouldTextInputAppearBelowOptions) {
+                defaultIndex = allOptions.length;
+            } else if (props.focusedIndex >= 0) {
+                defaultIndex = props.focusedIndex;
+            } else {
+                defaultIndex = props.selectedOptions.length;
```

Comment:
*regression:* if the length of selectedOptions is 0 then we get this bug https://github.com/Expensify/App/issues/37239. In the diff I posted above, we default it to -1 instead of 0.